Sources/SwiftDocC/SwiftDocC.docc/Resources/RenderNode.spec.json

```diff
@@ -2856,6 +2870,18 @@
                         }
                     }
                 }
+            },
+            "VersionPatch": {
+                "type": "object",
+                "required": ["version", "patch"],
+                "properties": {
+                    "version": {
+                        "$ref": "#/components/schemas/SchemaVersion"
```

Comment:
Should we require the version to be a specific format or just require this to be a unique string? 

I'm also wondering if we should have a separate display name?

For example, maybe I might want to use a commit hash as my unique version identifier, but then provide "Xenial" ([an ubuntu version name](https://en.wikipedia.org/wiki/Ubuntu_version_history#Ubuntu_16.04_LTS_(Xenial_Xerus))) as the display name.

Not entirely opposed to forcing Semantic Versioning here though- just curious what your thoughts are around this.

Reply:
I like that idea. I thought using Semantic Versioning would be pretty easy since it already exists in every RenderNode, so it would be easy to grab from there (and I was imagining it would be unique), but it seems more flexible to have a unique version identifier and a display name. And instead of having the user select two semantic versions to diff between, it sounds more readable to select two display names like Xenai and Yakkety.

Reply:
> Would the user have to pass all the version IDs and display names in on the command line then?

That's a good question. I think they we should allow for passing them via the command-line for automation use-cases but we already have a version property in the `Info.plist` file that can be included in a DocC catalog so I think we could use that as well: https://github.com/apple/swift-docc/blob/main/Sources/DocCDocumentation/DocCDocumentation.docc/Info.plist#L24.

And maybe add an additional `VersionDisplayName` property to that spec.